v2/scheduler/scheduler.go

```diff
@@ -140,6 +151,16 @@ func (s *scheduler) run(ctx context.Context) error {
 	// Wait for an error or a completed context
 	var err error
 	select {
+	// In essence, this comprises the Scheduler's "healthcheck" logic.
+	// Whenever we receive an error on this channel, we cancel the context and
+	// shut down.  E.g., if one loop fails, everything fails.
+	// This includes:
+	//   1. an error listing projects at the start of the project loop
+	//      (Scheduler <-> API comms)
+	//   2. an error instantiating a reader in any of the worker/job loops
+	//   3. an error instantiating a reader, reading a message or acking a
+	//      message in the healhcheck loop
+	//      (Scheduler <-> Messaging queue comms)
```

Comment:
Maybe worth explaining why we need no. 3, when no. 2 might seem to have the bases covered?

My sense is that no. 3 detects issues when the rest of the system is quiet and might not "naturally" encounter a problem.

Reply:
Added further context; indeed, your sense above is exactly it.

Reply:
👍